code/datums/datum.dm

```diff
@@ -31,6 +46,10 @@
 		if (timer.spent)
 			continue
 		qdel(timer)
+
+	for(var/target in observation_procs)
+		UnregisterObservation(target, observation_procs[target])
```

Comment:
Registering and unregistering should still be handled via the observation singletons

Reply:
remove